docs/2.develop/contracts/environment/environment.md

```diff
@@ -41,36 +41,32 @@ The `current_account` contains the address in which your contract is deployed. T
 
 ### Predecessor and Signer
 
-The `predecessor` is the account that called the method in the contract. Meanwhile, the `signer` is the account that _signed_ the first transaction that derived in such method call.
+The `predecessor` is the account that called the method in the contract. Meanwhile, the `signer` is the account that _signed_ the transaction to begin with.
```

Comment:
As a non english speaker, I prefer something closer to "that derived in such method call" than the succinct "to begin with".

Reply:
I'd go with:
```
The `predecessor` is the account that called the method in the contract. Meanwhile, the `signer` is the account that _signed_ the initial transaction.
```